bindings/pyroot/JupyROOT/kernel/rootkernel.py

```diff
@@ -37,8 +37,8 @@ def Debug(msg):
      print('Kernel main: %r' % msg, file=sys.__stderr__)
 
 class ROOTKernel(MetaKernel):
-    identifier_regex = r'[^\d\W](?:[\w]*(?:\.|->|::)?)*'
-    func_call_regex = r'([^\d\W](?:[\w]*(?:\.|->|::)?)*)\([^\)\()]*\Z'
+    identifier_regex = r'(?:\w(?:\w|\.|->|::|\d)*)'
+    func_call_regex = r'(?:\w(?:(?:\w|\.|->|::|\d))*)\([^\)\()]*\Z'
     implementation = 'ROOT'
     implementation_version = '1.0'
     language = 'c++'
```

Comment:
Do you think we could also add another commit with the removal of the creation of the Parser object? Since we saw it is redundant...